Doofinder/Feed/Controller/Adminhtml/Integration/CreateSearchEngine.php

```diff
@@ -0,0 +1,105 @@
+
+use Doofinder\Feed\Helper\Indexation;
+use Doofinder\Feed\Helper\StoreConfig;
+use Exception;
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\Escaper;
+use Magento\Store\Model\ScopeInterface;
+
+class CreateSearchEngine extends Action
+{
+    /** @var JsonFactory */
+    private $resultJsonFactory;
+
+    /** @var StoreConfig */
+    protected $storeConfig;
+
+    /** @var Escaper */
+    protected $escaper;
+
+
+    /**
+     * CleanIntegration constructor.
+     *
+     * @param JsonFactory $resultJsonFactory
+     * @param StoreConfig $storeConfig
+     * @param Context $context
+     */
+    public function __construct(
+        JsonFactory $resultJsonFactory,
+        StoreConfig $storeConfig,
+        Escaper $escaper,
+        Context $context
+    ) {
+        $this->resultJsonFactory = $resultJsonFactory;
+        $this->storeConfig = $storeConfig;
+        $this->escaper = $escaper;
+
+        parent::__construct($context);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute()
+    {
+        $resultJson = $this->resultJsonFactory->create();
+
+        $store = $this->storeConfig->getCurrentStore();
+
+        $storeGroupId = (int)$store->getStoreGroupId();
+        $installationId = $this->storeConfig->getValueFromConfig(StoreConfig::DISPLAY_LAYER_INSTALLATION_ID, ScopeInterface::SCOPE_GROUP, $storeGroupId);
+
+        $language = $this->storeConfig->getLanguageFromStore($store);
+        $currency = strtoupper($store->getCurrentCurrency()->getCode());
+
+        $storeId = (int)$store->getId();
+        $baseUrl = $store->getBaseUrl();
+
+        // store_id field refers to store_view's id.
+        $searchEngineConfig = [
+            "name" => $store->getGroup()->getName() . ' - ' . $store->getName(),
+            "language" => $language,
+            "currency" => $currency,
+            "site_url" => $baseUrl,
+            "callback_url" => $baseUrl . 'doofinderfeed/setup/processCallback?storeId=' . $storeId,
+            "options" => [
+                "store_id" => $storeId,
+                "index_url" => $baseUrl . 'rest/' . $store->getCode() . '/V1/'
+            ],
+            "store_id" => $installationId
+        ];
+        try {
+            $response =  $this->storeConfig->createSearchEngine($searchEngineConfig);
+
+            $this->storeConfig->setHashId($response["hashid"], $storeId);
+
+            $status = ["status" => Indexation::DOOFINDER_INDEX_PROCESS_STATUS_STARTED];
+            $this->storeConfig->setIndexationStatus($status, $storeId);
+            $resultJson->setData(['result' => $response]);
+        } catch (Exception $e) {
+
+            $resultJson->setData([
+                'result' => false,
+                'error' => $this->escaper->escapeHtml("Error creating search engine in Doofinder: " . $e->getMessage()),
+            ])->setHttpResponseCode(400);
```

Comment:
It's fine, but if you want an alternative (it's up to you :raised_hands: , although I would recommend to use the `use` statement to simplify this)

```suggestion
            ])->setHttpResponseCode(Magento\Framework\Webapi\Exception::HTTP_BAD_REQUEST);
```

Example of the simplification, seen in [another file](https://github.com/doofinder/doofinder-magento2/pull/385/files#diff-5e40f7c16fb71f11e1c442afa051998f629d9a7498613b8fbd43963c7cb2cf3a):

```php
use Magento\Framework\Webapi\Exception as WebapiException;

[...]

->setHttpResponseCode(WebapiException::HTTP_BAD_REQUEST);
```